e2e/dractions/retry.go

```diff
@@ -68,14 +68,32 @@ func waitDRPCPhase(ctx types.Context, namespace, name string, phase ramen.DRStat
 		}
 
 		currentPhase := drpc.Status.Phase
-		if currentPhase == phase {
-			log.Debugf("drpc \"%s/%s\" phase is %q in cluster %q", namespace, name, phase, hub.Name)
+		phaseMet := currentPhase == phase
+
+		currentProgression := drpc.Status.Progression
+		progressionCompleted := currentProgression == ramen.ProgressionCompleted
+
+		if phaseMet && progressionCompleted {
+			log.Debugf("drpc \"%s/%s\" phase is %q and progression is completed in cluster %q",
+				namespace, name, phase, hub.Name)
 
 			return nil
 		}
 
 		if time.Since(startTime) > util.Timeout {
-			return fmt.Errorf("drpc %q status is not %q yet before timeout in cluster %q, fail", name, phase, hub.Name)
+			msg := fmt.Sprintf("drpc \"%s/%s\" not in required phase and progression before timeout in cluster %q",
+				namespace, name, hub.Name)
+
+			if !phaseMet {
+				msg = msg + " " + fmt.Sprintf("required phase:%s, current phase:%s", phase, currentPhase)
+			}
+
+			if !progressionCompleted {
+				msg = msg + " " + fmt.Sprintf("required progression: %s, current progression: %s",
+					ramen.ProgressionCompleted, currentProgression)
```

Comment:
We got rid of these string concatenations and don't want to go back to this. I think what we do in waitDRPCReady() works well and is much simpler.